### PR TITLE
feat: add Wallet SDK Android docs

### PR DESCRIPTION
**Pull request by dom-murray**
## Description

Adding docs and API reference for the latest release of Wallet SDK Android (0.2.0)

## Resolved issues
Jira:

### Before submitting the PR, please consider the following:
- [ ] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems the pull request solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [x] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [x] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

**Comment by CassiusPacheco** [comment]
The docs themselves look good to me 👍 

Ideally we'd have a section apart for the Wallet SDKs? Currently it's being placed under the Core SDKs section. I dont mind merging it in for now and move it around later since we're not web devs, but double checking with @immutable/imx-developer-experience to see what they think?

Screen Shot 2022-08-12 at 3 28 49 pm
Screen Shot 2022-08-12 at 3 28 56 pm

File: docs/sdk-docs/wallet-sdk-android/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        appName = "My NFT Marketplace",
+        appDescription = "This is a marketplace where all my favorite NFTs can be traded.",
+        appIcons = listOf("http://www.marketplace.com/appicon.svg")
+    )
+)
+```
+If you want to use your own bridge server instead of the default provide it via `bridgeServerUrl` when connecting. For more info on how WalletConnect and the bridge works [see here](https://docs.walletconnect.com/tech-spec).
+
+### Handle callbacks
+All of the `ImmutableXWallet` methods (connect, disconnect etc) are asynchronous and changes to the status are communicated via the callback.
```
Comment (Jon-Alonso) [inline docs/sdk-docs/wallet-sdk-android/quickstart.md:29]:
```suggestion
All the `ImmutableXWallet` methods (connect, disconnect, etc.) are asynchronous, and changes to the status are communicated via the callback.
```

File: docs/sdk-docs/wallet-sdk-android/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            ImmutableXWalletStatus.Connecting -> { showProgressScreen() }
+            is ImmutableXWalletStatus.PendingConnection -> { showConnectPopup(status.intent) }
+            is ImmutableXWalletStatus.PendingSignature -> { showSignaturePopup(status.intent) }
+        }
+    }
+})
+```
+
+#### Pending states
+If a wallet app has been launched to connect or sign and your app has resumed but no result has arrived, PendingConnection or PendingSignature will be sent to the callback.
```
Comment (Jon-Alonso) [inline docs/sdk-docs/wallet-sdk-android/quickstart.md:47]:
```suggestion
If a wallet app has been launched to connect or sign and your app has resumed but no result has arrived, `PendingConnection` or `PendingSignature` will be sent to the callback.
```

File: docs/sdk-docs/wallet-sdk-android/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+This allows you to handle this scenario flexibly, you could use the provided intent to prompt the user to re-launch their wallet and complete the flow or continue showing a waiting state.
```
Comment (Jon-Alonso) [inline docs/sdk-docs/wallet-sdk-android/quickstart.md:49]:
```suggestion
This allows you to handle this scenario flexibly; you could use the provided intent to prompt the user to re-launch their wallet and complete the flow or continue showing a waiting state.
```

**Review by Jon-Alonso — APPROVED** [review]
@dom-murray LGTM, some minor grammar suggestions and a couple of questions:

1. We seem to be explicitly differentiating between Android (wallet) and Kotlin/JVM (core). I assume this is on purpose because core could technically be used outside of Android but I just wanted to confirm.

2. I'm not too familiar with the doc publishing process, but it looks like there are quite a few auto-gen HTML, CSS & js files. Can they be excluded from git? Or could we perhaps re-use some of these resources across docs?

**Comment by sprngbk** [comment]
Have a look at [this PR](https://github.com/immutable/imx-docs/pull/86). I've added a section for the Wallets SDKs on the SDK page and also updated the Switcher so Wallet SDK links can be passed in. Those pages just needed to be added so the links go to the right pages. See the `TODO` section in the PR description.

**Comment by dom-murray** [comment]
> @dom-murray LGTM, some minor grammar suggestions and a couple of questions:
> 
> 1. We seem to be explicitly differentiating between Android (wallet) and Kotlin/JVM (core). I assume this is on purpose because core could technically be used outside of Android but I just wanted to confirm.
> 2. I'm not too familiar with the doc publishing process, but it looks like there are quite a few auto-gen HTML, CSS & js files. Can they be excluded from git? Or could we perhaps re-use some of these resources across docs?

For 1 thats correct. For 2 I'd probably defer to @immutable/imx-developer-experience and how the want to handle versioning and static resources.

**Review by sprngbk — APPROVED** [review]
Looks good 👍

**Pull request merged**